### ISSUE TITLE
Improved first-time login experience

### ISSUE THREAD

**Issue by jshorland** [ushahidi/platform#1396]
- Begin every deployment with dummy data set only viewable by admin to help educate
- Create tutorial layer about how dummy data works and how to setup deployment
- Login by social signup?

Needs user testing.

**Comment by jshorland** [comment]
Assigning us all to indicate we'll need to scope, design and stage a user test for this

**Comment by jshorland** [comment]
Note: we need to have to have this in "dev up next" by end of quarter.

**Comment by brandonrosage** [comment]
I've put together the pieces to a first pass at scoping and designing the solution to this need.

**First**, is [a modest scope](https://docs.google.com/a/ushahidi.com/document/d/1AkFDYpYWMuwWfRB93GO81Afvq0wPvNYTQcfP-PcGRtg/edit?usp=sharing), which takes an iterative approach to rolling out an increasingly "smart" onboard tasking system.

Version 1 simply provides a step-by-step experience for on-boarding, with the option to "do this later." But there is not yet a systematic way to remind the user to return to the task(s) they skipped.

Version 2 uses email to periodically remind the user about tasks they haven't yet completed.

Version 3 uses email and Ushahidi's built-in Notifications ([yet to be completed](https://docs.google.com/a/ushahidi.com/document/d/15zejfhT7Tl4gjGO7cirD66L4Q4GOgiApZONFKvpkt1o/edit?usp=sharing)) to deliver these reminders.

**Second**, is [a semi-interactive wireframe](https://invis.io/WRABY6AVH) that illustrates what the user experience might look like. Highlights include:

### Language
I've front-loaded the "Language" control to the "Create new deployment" process so the subsequent on-boarding is done in users' chosen language. It should also help them understand which languages are supported before they commit to creating a deployment.

### Survey building
Including survey creation so early in the process underscores their value and importance to data collection. In this iteration, it gives them a blank slate. No placeholder survey or template.

### Data source configuration
Including data source configuration immediately after survey creation improves our communication about what they're for and how they're related. A noteworthy piece here, I think, is the treatment of the "survey form" as a data source alongside email, SMS, etc. By selecting any of the data sources not yet configured, their controls are displayed in a modal (e.g. try selecting "Email").

### User + role creation
There is just one question we're asking people in this step: "Who else is working with you?" There's not a separate question focused on roles. In this case, when you add a person, one of the controls is "Role," from which you can choose an available role or "Create new." By choosing "Create new," the "Roles" select control would be replaced by a fieldset for defining a new role:

![new role](https://cloud.githubusercontent.com/assets/1136279/22655012/4fe75b30-ec54-11e6-93be-dc7bc9ee572d.png)

The user could then define this new role or go back to using an existing role by selecting "Use an existing role," which would replace the fieldset with the previous select control.

### Share
When the user has reached the end of the on-boarding flow, they're delivered to the (empty) "Map" view, with a "You're all set!" message that prompts them to share their deployment. When this window is closed, they're dropped into their deployment with no more hand-holding measures.

----

In addition to any questions y'all might have, it'd be really helpful to get feedback from a few folks in particular:

* @jshorland: Both in scope and execution, does this seem reasonable to pull off before Q3?
* @rjmackay: Have I presented a solution that's reasonably within the technical boundaries of what we've built so far?
* @aoduor: Did I address all the touch points we flagged during our work in Aberdare?
* @caharding: How are you feeling about the flow I've presented here?

**Comment by aoduor** [comment]
I *really* dig this, @brandonrosage. As a first pass, i think this looks good. As we continue to iterate, it would be great to figure out
1. Where elements of training/showing them HOW to perform these tasks would fall into place. The simplified questions help quite alot.
2. How to tackle templating based on focus areas they select on deployment creation - in as much as we may not be able to handle that right now - but just make space for it?

That said, i think, looking at our initial commitment, this looks pretty good to me.

**Comment by brandonrosage** [comment]
**Note**: I've revised my summary above to include work I've added in order to outline Version 1: Intercom-powered on-boarding support. That work is included in the scope for the overall issue, as well as in [the addition of a UX concept](https://drive.google.com/open?id=1oDS_pZrX1TXq21Nqc7nEQRFt_UoeeIzITncmNbfho-o) for the Intercom-powered solution.

Example:

![Intercom example](https://cloud.githubusercontent.com/assets/1136279/22761114/002c8ff8-ee1f-11e6-8fbd-91bee55de9a7.png)

This was largely in response to @KrisKuri's comments in the Support & Engagement meeting Tuesday, as well as the "[Intercom strategy](https://drive.google.com/open?id=1s6DXAtY-uRMWIoLj1yYRVp8dA1DvM4nONZjRCNjZF6E)." Let's get this effort moving and fully-detailed.

**Comment by brandonrosage** [comment]
> Where elements of training/showing them HOW to perform these tasks would fall into place. The simplified questions help quite a lot, but we still need to show them how to do it. Maybe it'll be as simple as links, but we should think about this.

I think it'd be ideal for each of the messages I've drafted in [the UX concept](https://drive.google.com/open?id=1oDS_pZrX1TXq21Nqc7nEQRFt_UoeeIzITncmNbfho-o) to include links to relevant ushahidi.com/support pages (or embedded content, like videos). I haven't included it, however, because it's not clear what that content would be, how it's going to get created, or when. So short of those details, I designed a solution we could implement right away.

> How to tackle templating based on focus areas they select on deployment creation - in as much as we may not be able to handle that right now - but just make space for it?

In my mind, these considerations would be made for Version 3 or 4, where we know we have the ability to design solutions that require non-trivial developer support. I agree, we should save space for, say, pre-packaged surveys that deployers can choose from during the "Design your data" step. But I don't think we should design that until Version 3 or 4.

**Comment by caharding** [comment]
In my mind these onboading tips should be in addition to an in product tutorial, which will be much more helpful as a visual aid. These sorts of onboarding tips should only appear if people fail to complete essential functions I'd think.

**Comment by brandonrosage** [comment]
As I understand your comment, @caharding, you're saying that Version 1 (Intercom messaging) and Version 2 (immersive, step-by-step) need to co-exist -- that they can never stand alone in the absence of the other.

Is that what you're saying?

**Comment by brandonrosage** [comment]
[The Intercom-powered v1 on-boarding campaign is staged](https://app.intercom.io/a/apps/hl5rfiga/engage/messages/auto/campaigns/184844
).

Just [waiting on devs to configure the webhooks we nee](https://github.com/ushahidi/platform/issues/1595)d in order to send these messages to the right people at the right time.

**Comment by caharding** [comment]
Fantastic!

Charles Harding

Product Director

207-720-0713

On Fri, Mar 03, 2017 at 12:23 PM Brandon Rosage

<
mailto:Brandon Rosage <notifications@github.com>
> wrote:

a, pre, code, a:link, body { word-wrap: break-word !important; }

https://app.intercom.io/a/apps/hl5rfiga/engage/messages/auto/campaigns/184844
.

Just
https://github.com/ushahidi/platform/issues/1595
d in order to send these messages to the right people at the right time.

—

You are receiving this because you were mentioned.

Reply to this email directly,
https://github.com/ushahidi/platform/issues/1396#issuecomment-284060835
, or

**Comment by aoduor** [comment]
![yay](http://media2.giphy.com/media/2qPUYiMHrlkoU/giphy.gif)

**Comment by jshorland** [comment]
Related: #1891

**Comment by rjmackay** [comment]
@jshorland I have no idea whats needed on this issue anymore?

**Comment by rowasc** [comment]
Relevant context for https://github.com/ushahidi/platform-pattern-library/issues/205 but it appears the tasks listed were merged so closing in favor of the new UI refresh tasks
